lib/eslint.js

```diff
@@ -0,0 +1,243 @@
+#!/usr/src/app/bin/node_gc
+
+const fs = require("fs");
+const glob = require("glob");
+
+const eslint = require('./eslint-patch')();
+const docs = require('./docs')();
+const BatchSanitizer = require("./batch_sanitizer");
+const checks = require("./checks");
+const validateConfig = require("./validate_config");
+const computeFingerprint = require("./compute_fingerprint");
+const ConfigUpgrader = require("./config_upgrader");
+
+const CLIEngine = eslint.CLIEngine;
+const options = { extensions: [".js"], ignore: true, reset: false, useEslintrc: true };
+
+function run(print, runOptions) {
+  var configPath = runOptions.configPath || "/config.json";
+  var codeDir = runOptions.dir || "/code";
+
+  var cli; // instantiation delayed until after options are (potentially) modified
+  var debug = false;
+  var ignoreWarnings = false;
+  var ESLINT_WARNING_SEVERITY = 1;
+
+  // a wrapper for emitting perf timing
+  function runWithTiming(name, fn) {
+    const start = new Date();
+    const result = fn();
+
+    if (debug) {
+      const duration = (new Date() - start) / 1000;
+      console.error("eslint.timing." + name + ": " + duration + "s");
```

Comment:
This still feels fragmented to me. I'd go one of two ways:

- Pass `stdout` and `stderr` as constructor arguments, or
- Use `stdout` and `stderr` directly in the function and replace each in integration tests.

WDYT?